common/sampling.cpp

```diff
@@ -323,8 +269,10 @@ static llama_token_data_array llama_sample_probability_distribution_impl(
     // Get a pointer to the logits
     float * logits = llama_get_logits_ith(ctx_main, idx);
 
-    // Declare original_logits at the beginning of the function scope
-    std::vector<float> original_logits;
+    if (apply_grammar && original_logits != nullptr) {
+        // Only make a copy of the original logits if we are not applying grammar checks, not sure if I actually have to do this.
+        *original_logits = new std::vector<float>(logits, logits + llama_n_vocab(llama_get_model(ctx_main)));
```

Comment:
When is this freed?

Reply:
I added the line that deletes the allocated vector inside `llama_sampling_sample` in [fddd201](https://github.com/ggerganov/llama.cpp/pull/6240/commits/fddd201942f6b62e42ef3884496f735aa6163d3a)

Reply:
So the former code where I allocate/deallocated original_logits vector from different scopes seemed like bad practice so I made some adjustments in [27f2e85](https://github.com/ggerganov/llama.cpp/pull/6240/commits/27f2e855203e42b96bdb6adaff3adc051dc90fe0)